modules/afsocket/afunix-source.c

```diff
@@ -86,6 +87,14 @@ afunix_sd_init(LogPipe *s)
 {
   AFUnixSourceDriver *self = (AFUnixSourceDriver *) s;
 
+  if (is_file_socket(self->filename) == PU_FILE_TEST_FALSE)
+    {
+      msg_warning("Source is not a unix socket",
+                evt_tag_str("filename", self->filename),
+                NULL);
+      return FALSE;
+    }
+
```

Comment:
`bind()` is called during `afsocket_sd_init_method()` and this means that if `is_file_socket()` returns true, 
then the `bind()` with fail with EADDRINUSE

This is why I ask you to call this check _after_ the init method.